src/main/java/apoc/file/Neo4jFileStream.java

```diff
@@ -0,0 +1,83 @@
+package apoc.file;
+
+import apoc.ApocConfiguration;
+import apoc.util.FileUtils;
+import org.neo4j.procedure.*;
+import org.neo4j.logging.Log;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+public class Neo4jFileStream {
+    @Context
+    public Log log;
+
+    public static class FileEntry implements Comparable<FileEntry> {
+        public final long lineNo;
+        public final String line;
+        public final String path;
+
+        public FileEntry(long lineNumber, String data, String path) {
+            this.lineNo = lineNumber;
+            this.line = data;
+            this.path = path;
+        }
+
+        public int compareTo(FileEntry o) {
+            return new Long(this.lineNo).compareTo(o.lineNo);
+        }
+    }
+
+    public boolean canonicalPathInNeo4jHome(File f) throws IOException {
+        String canonicalPath = f.getCanonicalPath();
+        String neo4jHome = ApocConfiguration.get("unsupported.dbms.directories.neo4j_home", null);
+
+        return canonicalPath.contains(neo4jHome);
+    }
+
+    @Procedure(mode=Mode.DBMS)
+    @Description( "apoc.file.stream(path, { last: n }) - retrieve system or log file contents, optionally return only the last n lines" )
+    public Stream<FileEntry> stream(
+            @Name("path") String path,
+            @Name(value = "config",defaultValue = "{}") Map<String, Object> config) {
+        String neo4jHome = ApocConfiguration.get("unsupported.dbms.directories.neo4j_home", null);
```

Comment:
Seeing this repeated feels as if we might want to add a dedicated method to ApocConfiguration instead to retrieve the home.